core/src/main/java/kafka/server/share/SharePartition.java

```diff
@@ -2332,12 +2330,24 @@ String memberId() {
             return memberId;
         }
 
+        private RecordState rollbackState() {
```

Comment:
Seems the whole purpose exposing `rollbackState` is to verify if there is an ongoing transaction, correct? Then why to have state exposed rather have a method here `hasOngoingStateTransition()` which should return a boolean specifying if the state has any transaction in progress.